### PR TITLE
Build server on qt6 ci

### PR DESCRIPTION
**Pull request by nyalldawson**
A few real bugs here identified by the more stringent qt 6 requirements.

### CODE REVIEW

File: src/server/services/wfs/qgswfsdescribefeaturetype.cpp [hunk 1]
```diff
@@ -236,7 +236,8 @@ namespace QgsWfs
 
       //xsd:element
       QDomElement attElem = doc.createElement( QStringLiteral( "element" )/*xsd:element*/ );
-      attElem.setAttribute( QStringLiteral( "name" ), attributeName.replace( ' ', '_' ).replace( cleanTagNameRegExp, QString() ) );
+      const thread_local QRegularExpression sCleanTagNameRegExp( QStringLiteral( "(?![\\w\\d\\.-])." ) );
```
Comment (m-kuhn) [inline src/server/services/wfs/qgswfsdescribefeaturetype.cpp:239]: Is the former `cleanTagNameRegExp` something to reuse here (or delete)?

Comment (m-kuhn) [reply src/server/services/wfs/qgswfsdescribefeaturetype.cpp:239]: Forget it, it's gone already

File: src/server/services/wms/qgswmsutils.cpp [hunk 2]
```diff
@@ -157,7 +157,7 @@ namespace QgsWms
         break;
       default:
         QgsMessageLog::logMessage( QString( "Unsupported format string %1" ).arg( formatStr ) );
```
Comment (m-kuhn) [inline src/server/services/wms/qgswmsutils.cpp:159]:
```suggestion
        QgsMessageLog::logMessage( QStringLiteral( "Unsupported format string %1" ).arg( formatStr ) );
```

Unrelated thing spotted here

File: src/server/services/wms/qgswmsutils.cpp [hunk 3]
```diff
@@ -157,7 +157,7 @@ namespace QgsWms
...
-        saveFormat = UNKN;
+        saveFormat = "Unknown";
```
Comment (m-kuhn) [inline src/server/services/wms/qgswmsutils.cpp:160]: Is `UNKN` some define/variable that is still used somewhere or something that we can remove?

Comment (nyalldawson) [reply src/server/services/wms/qgswmsutils.cpp:160]: I think the "default:" here should actually be catching "UNKN" instead, but I'll let one of the server team confirm that...

**Review by m-kuhn — APPROVED** [review]
LGTM, went over it all and didn't see anything suspicious

**Comment by elpaso** [comment]
Nice to finally see some autos here.

**Comment by nyalldawson** [comment]
> Nice to finally see some autos here.

Ehh, it was by necessity, and I tried to offset it by removing some others. And I still feel dirty for adding them 🙃

File: src/server/services/wms/qgswmsutils.cpp (hunk 3 above)
Comment (nyalldawson) [reply src/server/services/wms/qgswmsutils.cpp:160]: Yep, it was supposed to be catching UNKN here. I've fixed that now.

**Pull request merged**